### PR TITLE
Use memory percentage for GC batch size

### PR DESCRIPTION
**Pull request by jmark99**
Update Accumulo to use a memory percentage value to determine the candidate batch size for garbage collection.

Updated Property.java to use a memory type rather than a bytes type.

Added logging to print the size and percentage used by the GC.

### CODE REVIEW

File: server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java [hunk 1]
```diff
@@ -90,7 +90,8 @@ public class SimpleGarbageCollector extends AbstractServer implements Iface {
     log.info("start delay: {} milliseconds", getStartDelay());
     log.info("time delay: {} milliseconds", gcDelay);
     log.info("safemode: {}", inSafeMode());
-    log.info("candidate batch size: {} bytes", getCandidateBatchSize());
+    log.info("GC candidate batch size: {} bytes ({} of GC memory)", getCandidateBatchSize(),
```
[the remark this answers is not in the document]
Comment (EdColeman) [reply server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java:93]: For a long running cluster, the logs will likely roll and that may be a consideration for keeping this - maybe a debug level?

[the remark this answers is not in the document]
Comment (ctubbsii) [reply server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java:93]: I think that's an unrelated change. We can add an independent thing if we want to report the config on a regular basis, or when something changes, or some other mechanism. I don't think that needs to be conflated with this PR.

[the remark this answers is not in the document]
Comment (ctubbsii) [reply server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java:93]: To be clear, I'm not suggesting we remove this log statement... just that the original message was sufficient. I'm trying to minimize unrelated changes to the core task of changing the property type. This is for a patch release, after all... minimizing changes to the stable version is the default goal.

**Comment by markowens** [comment]
Simplified PR to the single change in Property.java as per suggestion.

**Review by ctubbsii — APPROVED** [review]

**Pull request merged**